### PR TITLE
Refactored components, using recompose where necessary for anonymous fu…

### PR DESCRIPTION
**Pull request by Alemkin**
…nctions, removed dead code, etc

### CODE REVIEW

File: src/containers/Pager.js [hunk 1]
```diff
@@ -19,16 +20,24 @@ const mapDispatchToProps = dispatch => bindActionCreators({
...
     const nextPage = event.target.getAttribute('data-page')
     props.refreshData(props.name, {...props.filter, page: nextPage})
   }
 }
 
+const mapNumbersPagesHandlers = {
+  mapNumberedPagerButtons: props => (page, index) =>
+    <PagerButton loading={props.loading} key={index} page={page} currentPage={props.filter.page} changePage={props.changePage}>
+      {page + 1}
+    </PagerButton>
```
Comment (frodare) [inline src/containers/Pager.js]: I don't think we should include any UI logic in the containers.  I can think of two possible solutions:
1. create a PageButtons component and container that could more naturally handle this
2. transform this into a `getButtons` handler that would return a data object per button with which the component could map over

File: src/components/Status/index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-const determineFirstOnPage = (page, pageSize) => Math.max(page * pageSize, 0)
+const determineFirstOnPage = (page, pageSize) => Math.max(page * pageSize, 1)
```
Comment (frodare) [inline src/components/Status/index.js]: This seems wrong, why did you change it to a 1 starting index?

File: src/containers/Pager.js [hunk 3]
```diff
@@ -19,16 +20,24 @@ const mapDispatchToProps = dispatch => bindActionCreators({
...
+}
+
 const enhance = compose(
   connect(mapStateToProps, mapDispatchToProps),
-  withHandlers(handlers)
+  withHandlers(changePageHandlers),
+  withHandlers(mapNumbersPagesHandlers)
```
Comment (frodare) [inline src/containers/Pager.js]: I believe we should only need to wrap this once with `withHandlers` since it takes in an object

File: src/containers/Row.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { compose, withHandlers } from 'recompose'
+
+import Row from '../components/Row'
+
+const Cell = ({id, row}) => <td id={id}>{row[id]}</td>
+
```
Comment (frodare) [inline src/containers/Row.js]: All UI code should be removed from the container.  We probably need to discus this, maybe it is OK sometimes?

File: src/containers/Rows.js [hunk 5]
```diff
@@ -10,6 +12,13 @@ const mapStateToProps = (state, props) => {
   }
 }
 
-const mapDispatchToProps = dispatch => ({})
+const handlers = {
+  mapRow: props => row => <Row key={row.id} name={props.name} definition={props.definition} data={row} />
```
Comment (frodare) [inline src/containers/Rows.js]: UI code

File: src/containers/Pager.js (hunk 3 above)
Comment (Alemkin) [reply src/containers/Pager.js]: can't do that, the second function needs access to the first handler with props in it.

File: src/components/Status/index.js (hunk 2 above)
Comment (Alemkin) [reply src/components/Status/index.js]: from a user perspective, they expect to see 1-10, not 0-9
